pcdsdevices/gon.py

```diff
@@ -236,23 +252,344 @@ class Kappa(BaseInterface, Device):
+        without user's confirmation. Defaults to 2.
+
+    phi_max_step : int, optional
+        Maximum phi motor step, the largest move phi motor can make without
+        user's confirmation. Defaults to 2.
+
+    kappa_ang : number, optional
+        The angle of the kappa motor relative to the eta motor, in degrees.
+        Defaults to 50.
+
+    Notes
+    --------
+    When using the Kappa, it is most convenient to work through the pseudo
+    motors:
+
+    `kappa.e_eta`
+    `kappa.e_chi`
+    `kappa.e_phi`
+
+    Which have the normal motor functionalities (`mv`, `mvr`, `wm`).
+
+    It may be helpful to scan these pseudo motors to find the optimal position,
+    but make sure the step size is small enough so that you don't have to
+    confirm motion on every step.
+
+    Move commands will block the main thread, and pressing ctrl+c will cancel
+    motion.
+
+    The x, y, and z are the sample adjustment motors used to attain center of
+    rotation
+    """
     x = FCpt(IMS, '{self._prefix_x}', kind='normal')
     y = FCpt(IMS, '{self._prefix_y}', kind='normal')
     z = FCpt(IMS, '{self._prefix_z}', kind='normal')
+    # The real (or physical) positioners:
     eta = FCpt(IMS, '{self._prefix_eta}', kind='normal')
     kappa = FCpt(IMS, '{self._prefix_kappa}', kind='normal')
     phi = FCpt(IMS, '{self._prefix_phi}', kind='normal')
+    # The pseudo positioner axes:
+    e_eta = FCpt(PseudoSingleInterface, kind='normal', name='gon_kappa_e_eta')
+    e_chi = FCpt(PseudoSingleInterface, kind='normal', name='gon_kappa_e_chi')
+    e_phi = FCpt(PseudoSingleInterface, kind='normal', name='gon_kappa_e_phi')
 
     tab_component_names = True
 
-    def __init__(self, *, name, prefix_x, prefix_y, prefix_z, prefix_eta,
-                 prefix_kappa, prefix_phi, **kwargs):
+    tab_whitelist = ['stop', 'wait', 'k_to_e', 'e_to_k', 'mv_e_eta',
+                     'mv_e_chi', 'mv_e_phi', 'check_motor_step']
+
+    def __init__(self, *, name, prefix_x, prefix_y, prefix_z,
+                 prefix_eta, prefix_kappa, prefix_phi, eta_max_step=2,
+                 kappa_max_step=2, phi_max_step=2, kappa_ang=50, **kwargs):
         self._prefix_x = prefix_x
         self._prefix_y = prefix_y
         self._prefix_z = prefix_z
         self._prefix_eta = prefix_eta
         self._prefix_kappa = prefix_kappa
         self._prefix_phi = prefix_phi
+        self.eta_max_step = eta_max_step
+        self.kappa_max_step = kappa_max_step
+        self.phi_max_step = phi_max_step
+        self.kappa_ang = kappa_ang
         super().__init__('', name=name, **kwargs)
+
+    def stop(self):
+        """Stop the pseudo motors."""
+        self.eta.stop()
+        self.kappa.stop()
+        self.phi.stop()
+
+    def wait(self):
+        """Block until the action completes."""
+        self.eta.wait()
+        self.kappa.wait()
+        self.phi.wait()
+
+    @property
+    def eta_position(self):
```

Comment:
Are these being used anywhere? They seem redundant with `self.eta.wm()` and `self.eta.position`, for example

Reply:
I am only using them in `check_motor_step` and in the `mv_` functions - but I can just get it directly from `wm()` - they are redundant

Reply:
Took those out in f3ccec1